.github/workflows/ci-cpp-deb-pkg.yaml

```diff
@@ -0,0 +1,97 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: CI - CPP DEB packaging
+on:
+  pull_request:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/**'
+      - 'pulsar-client-cpp/**'
+  push:
+    branches:
+      - branch-*
+    paths:
+      - '.github/workflows/**'
+      - 'pulsar-client-cpp/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  cpp-deb-packaging:
+    name:
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: Tune Runner VM
+        uses: ./.github/actions/tune-runner-vm
+
+      - name: Detect changed files
+        id:   changes
+        uses: apache/pulsar-test-infra/paths-filter@master
+        with:
+          filters: .github/changes-filter.yaml
+          list-files: csv
+
+      - name: Check changed files
+        id: check_changes
+        run: echo "::set-output name=docs_only::${{ fromJSON(steps.changes.outputs.all_count) == fromJSON(steps.changes.outputs.docs_count) && fromJSON(steps.changes.outputs.docs_count) > 0 }}"
+
+      - name: Cache local Maven repository
+        if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/pulsar
+          key: ${{ runner.os }}-m2-dependencies-core-modules-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-m2-dependencies-core-modules-
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+        with:
+          distribution: 'temurin'
+          java-version: 17
+
+      - name: clean disk
+        if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+        run: |
+          sudo apt clean
+          docker rmi $(docker images -q) -f
+          df -h
+
+      - name: Package Pulsar source
+        if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+        run: mvn -B -ntp -q clean package -pl pulsar-client-api -am -DskipTests
+
+      - name: Build Debian packages
+        if: ${{ steps.check_changes.outputs.docs_only != 'true' }}
+        run: |
+          echo "Build Debian packages"
+          BUILD_IMAGE=1 pulsar-client-cpp/pkg/deb/docker-build-deb.sh
```

Comment:
Can we upload the built image to the docker hub?  This reduces build time.

Reply:
On one hand, see my PR description:

> Because currently only a few committers have the permission to
push the Docker images.

On the other hand, if these Dockerfiles (`pkg/deb/Dockerfile` and `pkg/rpm/Dockerfile`) changed, the changes can only be verified after uploading the images first. It's a manual operation and we cannot guarantee the docker image was built well. For example, if the docker image was built in an ARM64 architecture computer, it might not work well in the CI workflow.

Actually you can see it doesn't spend too much time building these images. I have already speeded up the build time for these Dockerfiles in the PR. In future, we can make other optimizations. If we use these images directly from Docker Hub, verifying the changes would be complicated and require some human operations.

<img width="842" alt="image" src="https://user-images.githubusercontent.com/18204803/184623301-fcb5726c-47a7-4558-bd1f-4646bba763d8.png">

Reply:
Ok, Thanks for your reply.